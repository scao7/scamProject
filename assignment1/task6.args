(lambda (x) (/ (if (= (% x 2) 0 ) 1.0 -1.0 ) ( + x 1)))
2
