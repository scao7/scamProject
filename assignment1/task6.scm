(define (square n)
	(* n n)
)

(define (S f n)
	(define i 0 )
	;(define sum 0.0)
	(define (iter sum i)
		(if(<= i n)
			(iter (+ sum (f i)) (+ i 1))
			sum
		)
	)
	(iter (f i) (+ i 1))
)

(define (w f n)
	(define i n)
	(cond 
		((= i 0 )(define result ( f i ) ))
		(else 
		(define a (S f i))
		(define b (S f (- i 1)))
		(define c (S f (+ i 1)))
		(define result (/(-(* c b) (square a)) (+(- c (* 2 a)) b)))
		)
	)
	result
)

(define (main)
	(setPort (open (getElement ScamArgs 1) 'read))
	(define arg (eval (readExpr) this))
	(define arg1 (readExpr))
	
	(println "(S " arg " " arg1 ") is " (fmt "%.15f" (S arg arg1)))
	(println "(w " arg " "arg1 ") is " (fmt "%.15f" (w arg arg1)))
)
