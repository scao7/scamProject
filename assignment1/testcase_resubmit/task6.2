(lambda (x) (* (if (= (% x 2) 0) 1 -1) (/ 1.0 (+ x 1)) (^ 0.5 (+ x 1))))
33
