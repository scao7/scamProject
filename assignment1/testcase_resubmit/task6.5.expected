(S <function anonymous(x)> 1356) is 0.693515504635209
(w <function anonymous(x)> 1356) is 0.693147180509973
